reference 9525, say subjectDN for EE should be null, close #35 (#53)
thomas-fossati/draft-tls13-iot: draft-ietf-uta-tls13-iot-profile.md

This section outlines the requirements for root CA certificates.

## subjectName

{{!RFC5280}} defines the subjectName field as follows: "The subject field identifies
the entity associated with the public key stored in the subject public key
field." RFC 5280 adds "If the subject is a CA then the subject field MUST be
populated with a non-empty distinguished name matching the contents of the
issuer field in all certificates issued by the subject CA."

> Root CA certificates MUST have a non-empty SubjectDN.

Author: ```suggestion
Root CA certificates MUST have a non-empty subjectName.
```

This section outlines the requirements for root CA certificates.

> ## Subject

Reviewer: ```suggestion
### Subject
```

Reviewer: need to increase indent

populated with a non-empty distinguished name matching the contents of the
issuer field in all certificates issued by the subject CA."

> Root CA certificates and Subordinate CA certificates MUST have a non-empty SubjectDN, as the value MUST match the DN of the Issuer.

Author: ```suggestion
Root CA certificates MUST have a non-empty SubjectDN.
```

populated with a non-empty distinguished name matching the contents of the
issuer field in all certificates issued by the subject CA."

> However, {{!RFC9525, Section 2.9}} now recommends that the SubjectDN be empty (null) for all End Entity certificates.

Author: ```suggestion
However, as {{!RFC9525, Section 2}} mandates that the subjectDN not be be used to identify a service, for IoT purposes, an empty SubjectDN avoids all confusion for End Entity certificates.
```

> Root CA and Subordinate CAs must have a non-null SubjectDN as that value must match the IssuerDN of subordinate certificates.

Reviewer 2: ```suggestion
Root CA certificates and Subordinate CA certificates MUST have a non-null SubjectDN, as the value MUST match the DN of the Issuer.
```

This section outlines the requirements for subordinate CA certificates.

> ### subjectName

Reviewer 2: ```suggestion
### Subject
```

> The subjectName field MUST be set and MUST contain the commonName, the organizationName,

Reviewer 2: ```suggestion
The subject field MUST be set and MUST contain the commonName, the organizationName,
```

This section outlines the requirements for end entity certificates.

> ### subjectName

Reviewer: ```suggestion
### Subject
```

{{!RFC9525, Section 2}} mandates that the subjectName not be be used to identify a service, for IoT purposes, an empty subjectName avoids all confusion for End Entity certificates.

The requirement in Section 4.4.2 of {{!RFC7925}} to only use EUI-64 for end

> entity certificates as a subjectName is lifted.

Reviewer 2: ```suggestion
entity certificates as a subject field is lifted.
```

Two fields are typically used to encode a device identifer, namely the
Subject and the subjectAltName fields. Protocol specifications tend to offer
recommendations what identifiers to use and the deployment situation is
fragmented.

> The subjectName field MAY include a unique device serial number. If the serial

Reviewer 2: ```suggestion
The subject field MAY include a unique device serial number. If the serial
```

### Subject

> {{!RFC9525, Section 2}} mandates that the subject field not be be used to identify a service, for IoT purposes, an empty subjectName avoids all confusion for End Entity certificates.

Reviewer: ```suggestion
{{!RFC9525, Section 2}} mandates that the subject field not be used to identify a service.
For IoT purposes, an empty subject field avoids all confusion for End Entity certificates.
```
